server/api/crud/pipelines.py

```diff
@@ -293,18 +294,24 @@ def _format_runs(
         self,
         runs: list[dict],
         format_: mlrun.common.formatters.PipelineFormat = mlrun.common.formatters.PipelineFormat.metadata_only,
+        kfp_client: mlrun_pipelines.utils.client = None,
     ) -> list[dict]:
         formatted_runs = []
         for run in runs:
-            formatted_runs.append(self._format_run(run, format_))
+            formatted_runs.append(self._format_run(run, format_, kfp_client))
         return formatted_runs
 
     def _format_run(
         self,
         run: PipelineRun,
         format_: mlrun.common.formatters.PipelineFormat = mlrun.common.formatters.PipelineFormat.metadata_only,
+        kfp_client=None,
     ) -> dict:
         run.project = self.resolve_project_from_pipeline(run)
+        if kfp_client:
+            error = self.get_error_from_pipeline(kfp_client, run)
+            if error:
+                run.error = error
```

Comment:
I think this should be moved to monitor runs

Reply:
I don't know if I would add it to monitor runs, it's already a heavy job and adding a get run request to pipelines for each step sounds not scaleable